File: tests/src/com.google.protobuf/protobuf-javalite/3.25.5/src/test/java/com_google_protobuf/protobuf_javalite/InternalTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertSame;

import com.google.protobuf.Empty;
import com.google.protobuf.Internal;
import org.junit.jupiter.api.Test;

public class InternalTest {
    @Test
    public void getDefaultInstanceReturnsGeneratedMessageDefaultInstance() {
        Empty expectedDefaultInstance = Empty.getDefaultInstance();

        Empty defaultInstance = Internal.getDefaultInstance(Empty.class);

        assertSame(expectedDefaultInstance, defaultInstance);
        assertEquals(0, defaultInstance.getSerializedSize());
    }
}

File: tests/src/com.typesafe.akka/akka-protobuf-v3_2.13/2.6.21/src/test/scala/com_typesafe_akka/akka_protobuf_v3_2_13/InternalTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_protobuf_v3_2_13

import akka.protobufv3.internal.Empty
import akka.protobufv3.internal.Internal
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test

class InternalTest {
  @Test
  def getDefaultInstanceUsesMessageStaticAccessor(): Unit = {
    val defaultInstance: Empty = Internal.getDefaultInstance(classOf[Empty])

    assertSame(Empty.getDefaultInstance, defaultInstance)
  }
}
